xkafka/middleware/slog.go

```diff
@@ -0,0 +1,39 @@
+package middleware
+
+import (
+	"context"
+	"time"
+
+	"log/slog"
+
+	"github.com/gojekfarm/xtools/xkafka"
+)
+
+// LoggingMiddleware is a middleware that logs messages using log/slog.
+func LoggingMiddleware(lvl slog.Level) xkafka.MiddlewareFunc {
```

Comment:
Can you make these variadic options?
```go
func LoggingMiddleware(lvl slog.Level) xkafka.MiddlewareFunc
```
```go
func LoggingMiddleware(opts ...Option) xkafka.MiddlewareFunc
```

Where, for example, a `slog.Handler` can also be passed; But if not, `slog.Default` is used.